samples/net/download/src/main.c

```diff
@@ -167,8 +167,27 @@ static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
 				       uint32_t event,
 				       struct net_if *iface)
 {
+	int err;
 	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
 		printk("Fatal error received from the connectivity layer, rebooting\n");
```

Comment:
This comment is not aligned with the functionality.

As a side note, does the sample recover from the modem reboot and reconnect?

Reply:
I tried to simulate the fault after the first packet is received by setting CFUN=0 and sending the modem error event and it does reconnect and continue the download in that case.